File: back/api/Models/Product/Review/ProductReview.cs
﻿using System.ComponentModel.DataAnnotations;

namespace api.Models.Product;

public class ProductReview
{
  [Key] public Guid Id { get; init; }
  [Required] public int ProductId { get; init; }
  [Required] public string RevieweesNickname { get; init; }
  [Required] public bool ByEmployee { get; init; }
  [Required] public string Title { get; init; }
  [Required] public string Content { get; init; }
  [Required] public int Stars { get; init; }
  [Required] public DateTimeOffset CreatedAt { get; init; }
  [Required] public bool IsDeleted { get; init; }

  public virtual IEnumerable<ProductReviewComment> Comments { get; init; }
}

File: apps/frontend/types/ProductReview.ts
import { Product } from "./Product";

export type ProductReview = {
  reviewId: string;
  productId: string;
  reviewersNickname: string;
  byEmployee: boolean;
  title: string;
  content: string;
  stars: number;
  createdAt: string;
};

export type ProductReviewWithProduct = ProductReview & {
  product: Product;
};

export type AddProductReviewRequestBody = Omit<
  ProductReview,
  "reviewId" | "productId" | "byEmployee" | "createdAt"
>;

export type ProductReviewComment = {
  reviewCommentId: string;
  reviewId: string;
  commentersNickname: string;
  byEmployee: string;
  title: string;
  content: string;
  createdAt: string;
};

export type AddProductReviewCommentRequestBody = Omit<
  ProductReviewComment,
  "reviewCommentId" | "reviewId" | "byEmployee" | "createdAt"
>;
